src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -68,39 +68,42 @@ public BufferReader(in ReadOnlySequence<byte> buffer)
         /// <summary>
         /// The unread portion of the <see cref="CurrentSpan"/>.
         /// </summary>
-        public ReadOnlySpan<byte> UnreadSpan
+        public ReadOnlySpan<T> UnreadSpan
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get => CurrentSpanIndex == 0 ? CurrentSpan : CurrentSpan.Slice(CurrentSpanIndex);
         }
 
         /// <summary>
-        /// The total number of bytes processed by the reader.
+        /// The total number of {T}s processed by the reader.
         /// </summary>
-        public int ConsumedBytes { get; private set; }
+        public int ConsumedValues { get; private set; }
 
         /// <summary>
-        /// Peeks at the next byte value without advancing the reader.
+        /// Peeks at the next T value without advancing the reader.
         /// </summary>
-        /// <returns>The next byte or -1 if at the end of the buffer.</returns>
+        /// <returns>The next T or default if at the end of the buffer.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public int Peek() => End ? -1 : CurrentSpan[CurrentSpanIndex];
+        public T Peek()
+        {
+            return End ? default : CurrentSpan[CurrentSpanIndex];
+        }
 
         /// <summary>
-        /// Read the next byte value.
+        /// Read the next T value.
         /// </summary>
-        /// <returns>The next byte or -1 if at the end of the buffer.</returns>
+        /// <returns>The next T or default if at the end of the buffer.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public int Read()
+        public T Read()
         {
             if (End)
             {
-                return -1;
+                return default;
```

Comment:
i think the method should be try read and return false instead of retuning default / -1

Reply:
I'm changing this to `bool Peek/Read(out T)`